python/ray/tests/test_advanced_2.py

```diff
@@ -745,6 +749,65 @@ def f():
     wait_for_pid_to_exit(pid1)
 
 
+# This case tests that the worker leaked issue when task finished with errors.
+#
+# Case steps are:
+#   1. Start a driver which creates a normal task with a long sleeping. This
+#      makes the normal task doesn't return.
+#   2. Send a SIGTERM to the normal task to trigger an error for it.
+#   3. After the normal task being reconstructed, we send a SIGTERM to the
+#      driver to make it offline and expects Ray collects the idle workers for
+#      the previous nomral task.
+def test_whether_worker_leaked_when_task_finished_with_errors(ray_start_regular):
+    driver_template = """
+import ray
+import os
+import ray
+import numpy as np
+import time
+from ray.experimental.internal_kv import _internal_kv_put
+
+ray.init(address="{address}", namespace="test")
+
+@ray.remote
+def normal_task(large):
+    # Record the pid of this normal task.
+    _internal_kv_put("NORMAL_TASK_PID", str(os.getpid()))
```

Comment:
Can you use an actor instead of internal KV? We shouldn't use unstable APIs for testing.

Reply:
I don't think a KVStore actor is the good practice. It  will make readers miss the key point of the test code.

Reply:
BTW, I found there're many places we use this internal KV for testing. :)

Reply:
Ah well, they shouldn't be using it either :) Could you still convert to using an actor here? It's standard to use actors like this in testing, and we should always prefer that over an unstable API.